<!--<div align="center">Семинар № 5-->

<div align="center"><font color=#4682B4 size=5 face=«Черное тело»><em><strong>Семинар № 5</strong></em></font></div>

<div class="alert alert-block alert-info"><em>
<b>ЗАДАЧА № 1.</b> Когда используется критерий Стьюдента, а когда Z –критерий?
</em></div>

<div class="alert alert-block alert-success"><em>
<b>Ответ:</b></em></div>

|<em>Z-критерий</em>|<em>критерий Стьюдента</em>|
|:------:|:------:|
|стандартное отклонение известно|стандартное отклонение неизвестно|
|размер выборки достаточно велик|размер выборки невелик|
|Нормальное распределение|Студент-т распределение|
|<font size="4px"> $Z_н=\frac{\bar X - \mu}{\sigma\ /\ \sqrt{n}}$|<font size="4px"> $t_н=\frac{\bar X - m}{\sigma_{несм}\ /\ \sqrt{n}}$|


<div class="alert alert-block alert-info"><em>
<b> ЗАДАЧА № 2. </b>Решать вручную.<br> Проведите тест гипотезы.<br>
Утверждается, что шарики для подшипников, изготовленные автоматическим станком,
имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05,
проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался
равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.</em></code>
</em></div>

<font size="2px">Сформулируем нулевую гипотезу $H_0:$<br>
<font size="3px">$\mu=\mu_0=17\ мм,$ <font size="2px">где<br> 
$\mu$ - средняя генеральной совокупности ;<br>
$\mu_0$ - средняя выборки .<br>
Тогда для правостороннего теста альтернативная гипотеза, $H_1:$<br>
<font size="3px">$\mu>\mu_0.$<br> 
<font size="2px">Поскольку нам известны и среднее и дисперсия генеральной совокупности,<br>
то для проверки гипотезы будем использовать $z$-критерий.<br>
<font size="4px"> $Z_н=\frac{\bar X - \mu_0}{\sigma\ /\ \sqrt{n}}$<font size="2px">, где<br>
$\bar X = \mu = 17,5\ мм$ - среднее по выборке;<br>
$\sigma = \sqrt{4} = 2\ мм$ - среднеквадратичное отклонение генеральной совокупности;<br>
$n = 100\ шт.$ - размер выборки.
<br><br>  
Для решения этой и следующих задач импортируем необходимые Python-библиотеки:

In [5]:
import numpy as np
import scipy.stats as st

In [6]:
z_nab = (17.5 - 17) / (np.sqrt(4) / np.sqrt(100))
print(f'Наблюдаемое значение z-критерия: {z_nab: .3f}')

Наблюдаемое значение z-критерия:  2.500


<font size="2px">Найдем табличное значение $z$ -критерия, для $\alpha = 0.05$:

In [7]:
z_tab=st.norm.ppf(1-0.05)
print(f'Табличное значение z-критерия: {z_tab: .4f}')

Табличное значение z-критерия:  1.6449


<font size="2px">Примем нулевую гипотезу, если <font size="3px">$Z_{наб}$ 
<font size="2px">принадлежит промежутку <font size="3px">$(-\infty\ ;\ 1,645)$

In [8]:
if z_nab > z_tab:
    print(f'Принимаем альтернативную гипотезу H1.')
else:
    print(f'Принимаем нулевую гипотезу.')    

Принимаем альтернативную гипотезу H1.


<div class="alert alert-block alert-success"><em>
<b>Ответ:</b><br>
<b>Отвергаем</b> нулевую гипотезу $H_0:\  \mu_0 = 17,0\ мм$ на уровне значимости
$\alpha = 0,05$.</em>
</div>

<div class="alert alert-block alert-info"><em>
<b> ЗАДАЧА № 3. </b>Решать вручную.<br> Проведите тест гипотезы.<br>
Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?<br>
(Провести двусторонний тест.)</em></code>

In [ ]:
<font size="2px">Сформулируем нулевую гипотезу $H_0:$<br>
<font size="3px">$\mu=200\ г,$ <font size="2px">где<br> 
$\mu$ - средняя генеральной совокупности ;<br>
Тогда для двустороннего теста $\alpha = (1-99\%)\ /\ 2=0,005$ и альтернативная гипотеза, $H_1:$<br>
<font size="3px">$\mu\not=200\ г.$<br> 
<font size="2px">Поскольку нам неизвестна дисперсия генеральной совокупности,<br>
то для проверки гипотезы будем использовать $t$-критерий Стьюдента.<br>
<font size="4px"> $t_н=\frac{\bar X - m}{\sigma_{несм}\ /\ \sqrt{n}}$<font size="2px">, где<br>
$\bar X$ - среднее по выборке;<br>
$\sigma_{несм}$ - несмещённая оценка дисперсии;<br>
$n = 10$ - размер выборки.
<br>
<br>

In [14]:
a = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
t_nab = (a.mean()-200)/(a.std(ddof=1)/np.sqrt(len(a)))
print(f'Наблюдаемое значение t-критерия: {t_nab: .3f}')

Наблюдаемое значение t-критерия: -1.065


In [16]:
t_tab1=st.t.ppf(1-0.01/2,len(a)-1)
t_tab2=st.t.ppf(0.01/2,len(a)-1)
print(f'Табличные значения t-критерия: {t_tab2: .4f}; {t_tab1: .4f}')

Табличные значения t-критерия: -3.2498;  3.2498


In [ ]:
<font size="2px">Примем нулевую гипотезу, если <font size="3px">$t_{наб}$ 
<font size="2px">принадлежит промежутку <font size="3px">$(-3.2498\ ;\ 3.2498)$

In [17]:
if t_nab > t_tab1 or t_nab < t_tab2:
    print(f'Отвергаем нулевую гипотезу H0.')
else:
    print(f'Принимаем нулевую гипотезу H0.')   

Принимаем нулевую гипотезу H0.


<div class="alert alert-block alert-success"><em>
<b>Ответ:<br> Принимаем</b> нулевую гипотезу $H_0:$ средний вес пачки печенья составляет 200 г на уровне значимости
$\alpha = 0,01$.
</em></div>

<div class="alert alert-block alert-info"><em>
<b> ЗАДАЧА № 4. </b>Решать с помощью функции.<br> Есть ли статистически значимые различия в росте дочерей?<br>
Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169, 165<br>
Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160</em></code>

<font size="2px">По идее, рост дочерей зависит от роста матерей, 
поэтому примем, что выборки зависимы.<br>
И решать будем с помощью функции <b>ttest_rel()</b> из Python-библиотеки.<br>
Приведём выборки к одной размерности, отбросив последнее наблюдение в выборке роста матерей.

In [24]:
b = np.array([172, 177, 158, 170, 178, 175, 164, 160, 169])
c = np.array([173, 175, 162, 174, 175, 168, 155, 170, 160])

<font size="2px">Сформулируем нулевую гипотезу $H_0:$<br>
На уровне значимости <b>$\alpha = 0,05$</b> статистически значимых различий в росте дочерей <b>НЕТ.</b>

In [21]:
st.ttest_rel(b, c)

Ttest_relResult(statistic=0.559522990335608, pvalue=0.5911212354055175)

<div class="alert alert-block alert-success"><em>
<b>Ответ:</b><br> Так как, $p-value=0.5911>\alpha=0.05$, нулевая гипотеза <b>принимается.</b><br>
Cтатистически значимых различий в росте дочерей НЕТ на уровне значимости 0,05.
</em></div>